core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -98,6 +97,7 @@ import scala.annotation.nowarn
  * Logic to handle the various Kafka requests
  */
 class KafkaApis(val requestChannel: RequestChannel,
+                val apisUtils: ApisUtils,
```

Comment:
Instantiating a "utils" object is a bit weird to me. it seems to me we can make ApisUtils be a ```trait``` with self-type (this: KafkaApis) if the main purpose of this PR is to reduce the size of ```KafkaApis```. The benefit of using self-type is that 
1. we don't need to instantiate a "utils"
1. we can move some code from KafkaApis to ApisUtils
1. we don't need to change "autxxx" to "apisUtils.autxxx"

Reply:
Thanks, @chia7712, I like this suggestion. I'll see if I can figure this out (I wasn't too happy with the utility class needing to be instantiated either)

Reply:
Is this an improvement? I am not sure. A couple of things:
1. The `Utils` suffix looks odd both in a trait as well as a class that is instantiated.
2. Using delegation allows for better separation of concerns than inheritance.

Reply:
1. Agreed. I don't really like the name, but can't think of a better one.

2. Originally, I had hoped to remove some dependencies from KafkaApis when pulling out these methods, but it turns out they are all needed for other things as well. In the end it just adds a new dependency (which is fine). Seeing how I had to mess around with the type hierarchy to not break or complicate TestRaftRequestHandler, I'm leaning towards the delegation approach we had before.